tsl/src/bgw_policy/job_api.c

```diff
@@ -87,8 +137,17 @@ job_add(PG_FUNCTION_ARGS)
 	namestrcpy(&proc_name, func_name);
 	namestrcpy(&owner_name, GetUserNameFromId(owner, false));
 
+	/* The check exists but does not have the expected signature: (config jsonb) */
+	if (OidIsValid(check) && !check_signature_is_valid(check))
+		ereport(ERROR,
+				(errcode(ERRCODE_INVALID_PARAMETER_VALUE),
+				 errmsg("function or procedure %s.%s(config jsonb) not found",
+						NameStr(check_schema),
+						NameStr(check_name)),
+				 errhint("The check function's signature must be (config jsonb).")));
```

Comment:
You are using this in two locations and in both you throw an error if the function is not found, so you can actually simplify the code and move the entire if-statement and the `ereport` into the `check_signature_is_valid` function and let it report the error there instead (and in that case call it something like `validate_signature`).

Reply:
+1